src/Header/ResponseParser.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+
+#ifndef GROUP4_SRC_RESPONSEPARSER_H_
+#define GROUP4_SRC_RESPONSEPARSER_H_
+
+#include <string>
+#include <memory>
+#include "OperatorData.h"
+
+// Used to make request string to OperateData(JSON like) format.
+// string responseStr = "[X: X_DATA, Y: Y_DATA]"
+// unique_ptr<OperateData> data = RequestParser::ParserHeader(responseStr)
+// data will return header and body.
+class ResponseParser
+{
+
+public:
+    
+    // Singleton code convention method
+    static unique_ptr<RequestParser> GetInstance();
```

Comment:
std::unique_ptr

Reply:
You should not return unique_ptr for singleton GetInstance(), because there can only be one instance of unique_ptr. Return a raw pointer instead.